qemu/tests/slof_balloon.py

```diff
@@ -0,0 +1,73 @@
+"""
+slof_balloon.py include following case:
+1. virtio balloon can work with pci-bridge
+"""
+import logging
+import time
+
+from virttest import error_context
+from provider import slof
+from virttest import utils_net
+
+
+def _wait_for_balloon_work(set_val, monitor, timeout=300, step=3.0):
+    """
+    Wait for the balloon value working.
+    """
+    end_time = time.time() + timeout
+
+    while time.time() < end_time:
+        if set_val == int(monitor.query('balloon')['actual']):
+            return True
+        else:
+            time.sleep(step)
+    return False
+
+
+@error_context.context_aware
+def run(test, params, env):
+    vm = env.get_vm(params["main_vm"])
+    content, _ = slof.wait_for_loaded(vm, test)
+
+    error_context.context("Check the output of SLOF.", logging.info)
+    slof.check_error(test, content)
+    error_context.context("Try to log into guest '%s'." % vm.name,
+                          logging.info)
+    timeout = float(params.get("login_timeout", 240))
+    session = vm.wait_for_login(timeout=timeout)
+    logging.info("log into guest '%s' successfully." % vm.name)
+
+    qmp_ports = vm.get_monitors_by_type('qmp')
+    if qmp_ports:
+        qmp_port = qmp_ports[0]
+    else:
+        test.error("Incorrect configuration, no QMP monitor found.")
+    error_context.context('Check virtio balloon device info.')
+    balloon_size = qmp_port.query('balloon')['actual']
+    logging.debug('The balloon size is %s' % balloon_size)
+    if int(balloon_size) != (int(params["mem"]) * 1024 * 1024):
+        test.error('The balloon size is not equal to %d'
+                   % (int(params["mem"]) * 1024 * 1024))
+
+    error_context.context('Change the balloon size to %s'
+                          % params['balloon_size'])
+    qmp_port.balloon(int(params['balloon_size']))
+    error_context.context('Check balloon size after changed.')
+    if not _wait_for_balloon_work(int(params['balloon_size']), qmp_port,
+                                  int(params['balloon_timeout'])):
```

Comment:
For 'balloon_timeout', the time that 'balloon' need will different for different balloon value, i.e, maybe timeout=300 is not ok for 'balloon value=200G',  so, it is better to specify a fixed value for "balloon value" for example 2g.